hummingbot/connector/derivative/binance_perpetual/binance_perpetual_api_order_book_data_source.py

```diff
@@ -59,59 +87,109 @@ async def get_last_traded_prices(cls, trading_pairs: List[str], domain: str = CO
+        return cls._trading_pair_symbol_map
+
+    @classmethod
+    async def init_trading_pair_symbols(
+        cls, domain: str = CONSTANTS.DOMAIN, throttler: Optional[AsyncThrottler] = None
+    ):
+        """Initialize _trading_pair_symbol_map class variable"""
+        api_factory = utils.build_api_factory()
+        rest_assistant = await api_factory.get_rest_assistant()
 
-    @staticmethod
-    async def fetch_trading_pairs(domain: str = CONSTANTS.DOMAIN, throttler: Optional[AsyncThrottler] = None) -> List[str]:
         url = utils.rest_url(path_url=CONSTANTS.EXCHANGE_INFO_URL, domain=domain)
-        throttler = throttler or BinancePerpetualAPIOrderBookDataSource._get_throttler_instance()
+        throttler = throttler or cls._get_throttler_instance()
         async with throttler.execute_task(limit_id=CONSTANTS.EXCHANGE_INFO_URL):
-            async with aiohttp.ClientSession() as client:
-                async with client.get(url=url, timeout=10) as response:
-                    if response.status == 200:
-                        data = await response.json()
-                        # fetch d["pair"] for binance perpetual
-                        raw_trading_pairs = [d["pair"] for d in data["symbols"] if d["status"] == "TRADING"]
-                        trading_pair_targets = [
-                            f"{d['baseAsset']}-{d['quoteAsset']}" for d in data["symbols"] if d["status"] == "TRADING"
-                        ]
-                        trading_pair_list: List[str] = []
-                        for raw_trading_pair, pair_target in zip(raw_trading_pairs, trading_pair_targets):
-                            trading_pair = utils.convert_from_exchange_trading_pair(raw_trading_pair)
-                            if trading_pair is not None and trading_pair == pair_target:
-                                trading_pair_list.append(trading_pair)
-                        return trading_pair_list
-        return []
+            request = RESTRequest(
+                method=RESTMethod.GET,
+                url=url,
+            )
+            response = await rest_assistant.call(request=request, timeout=10)
+
+            if response.status == 200:
+                data = await response.json()
+                # fetch d["pair"] for binance perpetual
+                cls._trading_pair_symbol_map = {
+                    d["pair"]: (f"{d['baseAsset']}-{d['quoteAsset']}")
+                    for d in data["symbols"]
+                    if d["status"] == "TRADING"
+                }
+
+    @staticmethod
+    async def fetch_trading_pairs(
+        domain: str = CONSTANTS.DOMAIN, throttler: Optional[AsyncThrottler] = None
+    ) -> List[str]:
+        OrderBookDataSource = BinancePerpetualAPIOrderBookDataSource
+        trading_pair_list: List[str] = []
+        symbols_map = await OrderBookDataSource.trading_pair_symbol_map(domain=domain, throttler=throttler)
+        trading_pair_list.extend(list(symbols_map.values()))
```

Comment:
The reason why this was done is that the class name is too long, often causing the line to be more than 120 characters long.

Reply:
ob_source_cls looks like a variable, not a class

Reply:
The "cls" postfix is used in several places around the code. You can find them by running
```
fgrep -rn --include "*.py" --include "*.pyx" --include "*.pxd" "_cls" hummingbot
```
in the terminal. The reason for this is because as introduced in a method, this is a variable that of type `type`. But in either case, I think we should do the re-definition at import instead like proposed in later comments:
```Python
from hummingbot.connector.derivative.binance_perpetual.binance_perpetual_api_order_book_data_source import (
    BinancePerpetualAPIOrderBookDataSource as BPOrderBookDataSource
)
```